driver-dse/src/main/java/com/datastax/driver/graph/GraphStatement.java

```diff
@@ -0,0 +1,135 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.graph;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import com.datastax.driver.core.SimpleStatement;
+import com.datastax.driver.core.exceptions.DriverException;
+
+/**
+ * A simple graph statement implementation.
+ */
+public class GraphStatement extends AbstractGraphStatement<SimpleStatement> {
+
+    private final String query;
+    private final Map<String, Object> valuesMap;
+    private List<String> JsonParams;
+    private int paramsHash;
+    private volatile ByteBuffer routingKey;
+
+    protected GraphStatement(String query, GraphSession session) {
+        super(session);
+
+        this.query = query;
+        this.valuesMap = new HashMap<String, Object>();
+        this.JsonParams = new ArrayList<String>();
+        this.routingKey = null;
+    }
+
+    public boolean hasValues() {
+        return this.valuesMap.size() > 0;
+    }
+
+    /*
+    Parameter values are supposed to be sent as JSON string in a particular format.
+    The format is : {name":"parameterName", "value":parameterValue}
+     */
+    private void processValues() {
+        JsonNodeFactory factory = new JsonNodeFactory(false);
+        JsonFactory jsonFactory = new JsonFactory();
+        ObjectMapper objectMapper = new ObjectMapper();
+        if (this.paramsHash == this.valuesMap.hashCode()) {
+            // Avoids regenerating the Json params if the params haven't changed.
+            return;
+        }
+        this.JsonParams.clear();
```

Comment:
But `processValues` is only ever called when we execute the statement, so that can be done at the last moment (could even be done in the GraphSession code). I agree that this should not be an instance member.